src/RazorSdk/Tool/ServerProtocol/ServerConnection.cs

```diff
@@ -274,6 +274,40 @@ public static string GetTempPath(string workingDir)
             }
         }
 
+        private static string FindDotNetExecutable()
+        {
+            var expectedPath = Environment.GetEnvironmentVariable("DOTNET_HOST_PATH");
+            if (!string.IsNullOrEmpty(expectedPath))
+            {
+                return expectedPath;
+            }
+
+#if NET
+            expectedPath = System.Environment.ProcessPath;
+#else
+            expectedPath = Process.GetCurrentProcess().MainModule.FileName;
+#endif
+
+            if ("dotnet".Equals(Path.GetFileNameWithoutExtension(expectedPath)))
+            {
+                return expectedPath;
+            }
+
+            // We were probably running from Visual Studio or Build Tools and found MSBuild instead of dotnet. Use the PATH...
+            var paths = Environment.GetEnvironmentVariable("PATH").Split(Path.PathSeparator);
+            var exeName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "dotnet.exe" : "dotnet";
+            foreach (string path in paths)
+            {
+                var dotnetPath = Path.Combine(path, exeName);
+                if (File.Exists(dotnetPath))
+                {
+                    return dotnetPath;
+                }
+            }
+
+            return "dotnet";
```

Comment:
```suggestion
            return exeName;
```

Reply:
Can you explain why you think it's necessary to use exeName rather than "dotnet"? I think either should work. From a functional perspective, the difference lies in whether the extension is included. I chose "dotnet" because that's [how it was before my original change](https://github.com/dotnet/sdk/blame/4ef9128d136cfc9e55c0374d01b117a9790c78a6/src/RazorSdk/Tool/ServerProtocol/ServerConnection.cs#L291), and reverting to previous behavior seems safest to me.

Reply:
It's less precise on Windows. It would for example allow invocation of `dotnet.cmd` which seems wrong. My mind starts asking questions like is that spoofable?

Reply:
If we've reached this code path, we've already given up on being truly safe. That said, some users may be very cautious about which executables they download without noticing dotnet.cmd, so I could see this making the user just slightly safer. I can make the change, thanks.